gobblin-runtime/src/main/java/gobblin/runtime/api/GobblinInstanceDriver.java

```diff
@@ -22,8 +22,14 @@
  * {@link JobSpec}s and run them.
  * */
 @Alpha
-public interface GobblinInstanceDriver extends Service {
+public interface GobblinInstanceDriver extends Service, JobLifecycleListenersContainer {
   JobCatalog getJobCatalog();
+  /**
+   * Returns a mutable instance of the job catalog
+   * (if it implements the {@link MutableJobCatalog} interface). Implementation will throw
+   * ClassCastException if the current catalog is not mutable.
+   */
+  MutableJobCatalog getMutableJobCatalog();
```

Comment:
The catalog is not optional. If someone is trying to use an Immutable catalog as mutable, that's an error.